dbt/adapters/databricks/credentials.py

```diff
@@ -241,172 +240,150 @@ def extract_cluster_id(cls, http_path: str) -> Optional[str]:
     def cluster_id(self) -> Optional[str]:
         return self.extract_cluster_id(self.http_path)  # type: ignore[arg-type]
 
-    def authenticate(self, in_provider: Optional[TCredentialProvider]) -> TCredentialProvider:
+    def authenticate(self) -> "DatabricksCredentialManager":
         self.validate_creds()
-        host: str = self.host or ""
-        if self._credentials_provider:
-            return self._provider_from_dict()  # type: ignore
-        if in_provider:
-            if isinstance(in_provider, m2m_auth) or isinstance(in_provider, token_auth):
-                self._credentials_provider = in_provider.as_dict()
-            return in_provider
-
-        provider: TCredentialProvider
-        # dbt will spin up multiple threads. This has to be sync. So lock here
-        self._lock.acquire()
-        try:
+        assert self._credentials_manager is not None, "Credentials manager is not set."
+        return self._credentials_manager
```

Comment:
is lock still required?

Reply:
I am just honoring the existing behavior. If there are multiple connections using the same DatabricksCredentials instance, there is a shared resource _config which can have concurrent write. @benc-db I think you introduce the original lock, is it still needed?

Reply:
I don't think so, but not totally sure.  I think I brought this lock over from somewhere else (from code Ray wrote, which was probably cribbed in part from dbt-spark code).